### PR TITLE
[code-clean] Use Go 1.21 `maps` package to clone and copy maps

### PR DESCRIPTION
**Pull request by iholder101**
In this PR I'm starting to use new Go's `maps` package that's presented in Go 1.21.
This package leverages generics in order to provide official functions to clone, copy, delete and check equality, and more.

### What this PR does
Before this PR:
We manually copy and clone maps with code that is duplicated throughout the repo.

After this PR:
Duplicated private functions are deleted and official functions from the `maps` package are being used.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- ~[ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required~
- [X] PR: The PR description is expressive enough and will help future contributors
- [X] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- ~[ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required~
- [X] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- ~[ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.~
- ~[ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered~

### Release note

```release-note
NONE
```

/sig code-quality

### CODE REVIEW

**Comment by enp0s3** [comment]
@iholder101 Thanks!
/approve

**Review by nunnatsa — APPROVED** [review]
Love this!

File: pkg/storage/types/dv.go [hunk 1]
```diff
@@ -111,22 +112,9 @@ func GenerateDataVolumeFromTemplate(clientset kubecli.KubevirtClient, dataVolume
...
-		annotations[k] = v
-	}
-
-	// passed to PVC by DataVolume controller
-	annotations[allowClaimAdoptionAnnotation] = "true"
-
-	newDataVolume.ObjectMeta.Annotations = annotations
+	newDataVolume.ObjectMeta.Labels = maps.Clone(dataVolumeTemplate.Labels)
+	newDataVolume.ObjectMeta.Annotations = maps.Clone(dataVolumeTemplate.Annotations)
+	newDataVolume.ObjectMeta.Annotations[allowClaimAdoptionAnnotation] = "true"
```
[the remark this answers is not in the document]
Comment (iholder101) [reply pkg/storage/types/dv.go:117]: Wow.. that's a shame..

So in order to create a map clone and initialize a map, we now need to:
```go
newDataVolume.ObjectMeta.Annotations = maps.Clone(dataVolumeTemplate.Annotations)
if newDataVolume.ObjectMeta.Annotations == nil {
	newDataVolume.ObjectMeta.Annotations = make(map[string]string)
}
newDataVolume.ObjectMeta.Annotations[allowClaimAdoptionAnnotation] = "true"
```

Oof...
I'm not gonna lie, I was expecting something more elegant from Go, especially after waiting so long for it. Seems that go developers just LOVE boilerplate.

I'll fix it. Thanks @nunnatsa!

File: pkg/storage/types/dv.go [hunk 2]
```diff
@@ -111,22 +112,15 @@ func GenerateDataVolumeFromTemplate(clientset kubecli.KubevirtClient, dataVolume
 	newDataVolume := &cdiv1.DataVolume{}
 	newDataVolume.Spec = *dataVolumeTemplate.Spec.DeepCopy()
 	newDataVolume.ObjectMeta = *dataVolumeTemplate.ObjectMeta.DeepCopy()
-
-	labels := map[string]string{}
-	for k, v := range dataVolumeTemplate.Labels {
-		labels[k] = v
+	newDataVolume.ObjectMeta.Labels = maps.Clone(dataVolumeTemplate.Labels)
+	if newDataVolume.ObjectMeta.Labels == nil {
+		newDataVolume.ObjectMeta.Labels = make(map[string]string)
```
Comment (nunnatsa) [inline pkg/storage/types/dv.go:117]: @iholder101 WDYT about this (not sure it's better, but I think it's maybe more readable)?
```suggestion

	if dataVolumeTemplate.Labels == nil {
		newDataVolume.ObjectMeta.Labels = make(map[string]string)
	} else {
		newDataVolume.ObjectMeta.Labels = maps.Clone(dataVolumeTemplate.Labels)
	}	
```

**Pull request merged**